### PR TITLE
[#187364733] -Ft two factor authentication for sellers

### PR DESCRIPTION
**Pull request by EricNiyo73**
**What does this PR do?**
This PR introduces the functionality for users, specifically Sellers, to enable two-factor authentication (2FA) for added security when logging into the application.

 **Description of Task to be completed?**
The task involves implementing the logic to allow users to enable 2FA for their accounts. When 2FA is enabled for a user, they should be prompted to provide an additional authentication factor, such as an token and OTP code sent via email, during the login process.

 **How should this be manually tested?**
Attempt to log in to the user account with 2FA enabled.
Verify that the login process prompts for a 2FA token i.e jwt token and OTP code sent via email of a user.
Enter a valid 2FA token and submit the form.

 **Any background context you want to provide?**
Verify that the user is successfully logged in after the 2FA token is validated.
Implementing 2FA adds an extra layer of security to user accounts, especially for users with elevated privileges like Sellers who may have access to sensitive data or functionalities.

 **What are the relevant pivotal tracker/Trello stories?**
[#187364733]
 **Screenshots (if appropriate)**
1) A user try to login with valid credentials, if a userRole is a seller, might get this response.
image

2) OTP sent and well received via email.
image

3) A user if tries to insert OTP  token which is not provided and also that OTP does not corresponded to his/her email 
image

4) A user finally entered a valid OTP code and authenticated successfully and also token would be generated for login.

image

### CODE REVIEW

File: test/jwtConfiguration.test.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (AngeloChristian1) [inline test/jwtConfiguration.test.ts]: Try to remove these eslint comments

File: package.json [hunk 2]
```diff
@@ -7,8 +7,8 @@
     "start": "tsc && node dist/index.js",
     "dev": "nodemon --exec ts-node index.ts --watch src --ext ts,js",
     "build": "tsc",
-    "test": "nyc mocha --require ts-node/register test/*.ts",
-    "coverage": "NODE_ENV=test nyc --reporter=text --reporter=lcov mocha --require ts-node/register test/*.ts --coverage && nyc report --reporter=text-lcov | coveralls",
+    "test": "nyc mocha --require ts-node/register test/*.ts --timeout 800000 --exit",
```
Comment (AngeloChristian1) [inline package.json]: why did you change the test command?
you can just add the timeout on the individual test you want to time

File: src/assets/emailTemplates/index.ts
Comment (Furaha-Justine) [inline src/assets/emailTemplates/index.ts:1]: I think you can rename this file to something else for better understanding.

File: src/docs/signupDocs.ts
Comment (Furaha-Justine) [inline src/docs/signupDocs.ts:1]: Also for better structure of project you can create new file for this doc.

**Review by Furaha-Justine — CHANGES_REQUESTED** [review]

File: .codeclimate.yml
Comment (AngeloChristian1) [inline .codeclimate.yml:1]: can you create a fix PR for this file

File: src/utils/hashPassword.ts
Comment (AngeloChristian1) [inline src/utils/hashPassword.ts:1]: there is another similar file in the utils named password.ts, This is not needed

Comment (AngeloChristian1) [inline src/utils/hashPassword.ts:1]: Hey, so there is a similar file in the rebased content named passwords.ts and it is the one we are using

**Review by Furaha-Justine — APPROVED** [review]

**Review by Furaha-Justine — APPROVED** [review]

**Review by AngeloChristian1 — APPROVED** [review]

**Review by AngeloChristian1 — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**